recipe/build.sh

```diff
@@ -1,24 +1,20 @@
 #!/bin/bash
 
 if [[ $HOST == *apple* ]]; then
-    # Apply patch to build.py.in (autoreconf might not be needed?)
-    patch -p1 < ${RECIPE_DIR}/osx_patch.diff
-    # These two flags are mentioned in the LHAPDF docs
+    # Bug for macOS as of lhapdf v6.1.1+
+    # https://gitlab.com/hepcedar/lhapdf/-/blob/c5b048f5794e483d85801d1ae39f42aaa46999dd/INSTALL#L162-164
     export CFLAGS=-Qunused-arguments
     export CPPFLAGS=-Qunused-arguments
-fi
 
-if [[ $HOST == *arm64* ]]; then
-    # Get an updated config.sub and config.guess
-    cp $BUILD_PREFIX/share/gnuconfig/config.* ./config/
-    # And autoreconf!
-    autoreconf -i -f
-    (cd wrappers/python && cython --cplus lhapdf.pyx)
+    autoreconf --install --force
+    (cd wrappers/python && cython --include-dir . --cplus lhapdf.pyx)
```

Comment:
Are these two needed in non-arm apple? (because they were not necessary before)

Reply:
I can't find a good diff to compare with, so let me push a commit that breaks this out into a `arm64` block again and see.

Reply:
Okay, thanks!

I believe the closer this is to the "default" installation in as many systems as possible, the easier it will be to accommodate future changes.